content/open-source/wolfi/building-a-wolfi-package.md

```diff
@@ -0,0 +1,437 @@
+
+- Downloading Packages
+  - `fetch`
+  - `git-checkout`
+- Autoconf
+  - `autoconf/configure`
+  - `autoconf/make`
+  - `autoconf/make-install`
+- Cmake
+  - `cmake/build`
+  - `cmake/configure`
+  - `cmake/install`
+- Go
+  - `go/build`
+  - `go/install`
+- Meson
+  - `meson/compile`
+  - `meson/configure`
+  - `meson/install`
+- Ruby
+  - `ruby/build`
+  - `ruby/clean`
+  - `ruby/install`
+- Split
+  - `split/debug`
+  - `split/dev`
+  - `split/infodir`
+  - `split/locales`
+  - `split/manpages`
+  - `split/static`
+- Other
+  - `strip`
+  - `patch`
+
+Each pipeline can have one or more parameters that should be provided as keypairs in a `with` entry. For example, this is how a download-and-check step looks like in the melange YAML, using the built-in pipeline `fetch`:
+
+```yaml
+  - uses: fetch
+    with:
+      uri: https://libzip.org/download/libzip-${{package.version}}.tar.gz
+      expected-sha256: 52a60b46182587e083b71e2b82fcaaba64dd5eb01c5b1f1bc71069a3858e40fe
+```
+
+Naturally, you can also run raw bash commands in your pipeline. The following example shows the build of [Composer](https://getcomposer.org/), the PHP package manager, which is not compiled as a typical package:
+
+```yaml
+  - name: Install Composer
+    runs: |
+      EXEC_DIR="${{targets.destdir}}/usr/bin"
+      mkdir -p "${EXEC_DIR}"
+      mv composer.phar "${EXEC_DIR}/composer"
+      chmod +x "${EXEC_DIR}/composer"
+```
+
+As indicated, a pipeline step will have either a `uses` or a `run` directive. You can have as many steps as you need, and you can use special variable substitutions inside steps:
+
+- `${{package.name}}`: Package name.
+- `${{package.version}}`: Package version.
+- `${{package.epoch}}`: Package epoch.
+- `${{targets.destdir}}`: Directory where final package files will be stored. Everything that lives here will be packed into your final apk.
+- `${{targets.subpkgdir}}`: Directory where final subpackage files will be stored. Works the same way as `targets.destdir`, but for subpackages.
+
+You can find more details about available pipelines in the [melange pipelines documentation](https://edu.chainguard.dev/open-source/melange/melange-pipelines).
+
+### The `subpackages` Section
+As mentioned above, a package may extract parts of its contents into subpackages, in order to make for a slimmer final apk. Many packages have resources that are not required at execution time: development headers, man pages, shared libraries that are optional… you name it. This part is really important in Wolfi, because we want packages to be minimal. The `subpackages` section of the melange YAML file looks a lot like the pipeline section, and it essentially works the same way. You'll just have to make sure you place any subpackage files in the `targets.subpkgdir` location.
```

Comment:
```suggestion
As mentioned previously, a package may extract parts of its contents into subpackages in order to make for a slimmer final apk. Many packages have resources that are not required at execution time, including development headers, man pages, shared libraries that are optional. This part is really important in Wolfi, because we want packages to be minimal. The `subpackages` section of the melange YAML file looks a lot like the pipeline section, and it essentially works the same way. You'll just have to make sure you place any subpackage files in the `targets.subpkgdir` location.
```

The ellipses seem a little informal to me